### PR TITLE
[Bug] [API] queryProcessDefinitionByCode bug

### PR DESCRIPTION
**Pull request by brave-lee**
#7220 
queryProcessDefinitionByCode bug
scheduler update api bug

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TaskDefinitionLogMapper.xml [hunk 1]
```diff
@@ -47,6 +47,9 @@
                 and version = #{item.version})
             </foreach>
         </if>
+        <if test="taskDefinitions == null or taskDefinitions.size == 0">
```
Comment (zhuangchong) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TaskDefinitionLogMapper.xml]: Judge taskDefinitions as a required parameter, please add it to the business layer.

**Review by zhuangchong — CHANGES_REQUESTED** [review]

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TaskDefinitionLogMapper.xml (hunk 1 above)
Comment (brave-lee) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TaskDefinitionLogMapper.xml]: OK

**Review by lenboo — APPROVED** [review]
+1

**Review by zhuangchong — APPROVED** [review]
+1

**Pull request merged**